### PR TITLE
Use ResumingBigtableResultScanner to handle rescan.

### PR DESCRIPTION
**Pull request by kevinsi4508**
Use ResumingBigtableResultScanner for rescan.

Rename the orginal ResumingBigtableResultScanner to RetryingStreamingResultScanner as it is for retry on errors.

### CODE REVIEW

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java [hunk 1]
```diff
@@ -238,28 +240,67 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
...
-  private ResultScanner<Row> readRows(ReadRowsRequest request, boolean resumable) {
-    // Delegate all resumable operations to the scanner. It will request a non-resumable
-    // scanner during operation.
-    if (resumable) {
-      return new ResumingStreamingResultScanner(
-        retryOptions,
-          request,
-          new BigtableResultScannerFactory() {
+  @SuppressWarnings("javadoc")
+  private static boolean hasAdaptedWhileMatchFilter(RowFilter filter) {
```
Comment (AngusDavis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: Drive-by code review comment: I'd argue that this class / package shouldn't have any notion of HBase concepts like WhileMatchFilters. Especially since label transformers could be applied by users of a lightweight client (this thin wrapper over grpc).

Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: That is fair. Let me update the code a bit. thanks!

Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: Overloaded readRows(), let me know if you know a better option.

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataClient.java [hunk 2]
```diff
@@ -84,11 +84,16 @@ CheckAndMutateRowResponse checkAndMutateRow(CheckAndMutateRowRequest request)
...
   /**
-   * Perform a scan over rows.
+   * Perform a scan over rows with resumable defaults to {@code false}.
    */
   ResultScanner<Row> readRows(ReadRowsRequest request);
 
   /**
+   * Perform a scan over rows with resumable option.
+   */
+  ResultScanner<Row> readRows(ReadRowsRequest request, boolean resumable);
```
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataClient.java]: Can we figure out if the request needs to be resumable?

Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataClient.java]: yes

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java [hunk 3]
```diff
@@ -238,28 +240,42 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
...
    */
-  private ResultScanner<Row> readRows(ReadRowsRequest request, boolean resumable) {
-    // Delegate all resumable operations to the scanner. It will request a non-resumable
-    // scanner during operation.
-    if (resumable) {
-      return new ResumingStreamingResultScanner(
-        retryOptions,
-          request,
-          new BigtableResultScannerFactory() {
+  private ResultScanner<Row> readRows(
```
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: I'm not a fan of this method.  It did 2 things instead of a single thing, and now it does 3.  I'd rather this be broken up into three methods that each add their own

Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: done.

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java [hunk 4]
```diff
@@ -238,31 +240,65 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
...
+          }
+        });
+  }
+
+  /**
+   * Returns a resumable scanner needed for rescan.
+   */
+  private ResultScanner<Row> getResumableScanner(ReadRowsRequest request) {
+    return  new RetryingStreamingResultScanner(
+      retryOptions, request, new BigtableResultScannerFactory() {
```
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: Can we create a member variable for BigtableResultScannerFactory?  Can we change the implementation to call getNonresumableScanner() [or what ever other name we chose]?

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java [hunk 5]
```diff
@@ -238,31 +240,65 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
...
+        public ResultScanner<Row> createScanner(ReadRowsRequest request) {
+          return readRows(request, false, false);
+        }
+      });
+  }
+
+  /**
+   * Returns a non-resumable streaming scanner.
+   */
+  private ResultScanner<Row> getNonresumableScanner(ReadRowsRequest request) {
```
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: Can we call this something that has a more positive name?  Something like createStreamingScanner?

Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: done

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataClient.java (hunk 2 above)
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataClient.java]: So can we move the logic of figuring that out into BigtableDataClient?

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java (hunk 4 above)
Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: done.
